models/comments-model.js

```diff
@@ -0,0 +1,47 @@
+const db = require("../db/connection");
+const { selectArticles } = require("../models/articles-model");
+
+exports.selectComments = (articleId) => {
+  if (!Number.isInteger(Number(articleId))) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  return db
+    .query(
+      "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC",
+      [articleId]
+    )
+    .then((result) => {
+      const comments = result.rows;
+      if (comments.length === 0) {
+        return Promise.reject({ status: 404, msg: "Page not found" });
+      }
+      return comments;
+    });
+};
+
+exports.createComments = (articleId, { username, body }) => {
+  if (
+    !Number.isInteger(Number(articleId)) ||
+    typeof username !== "string" ||
+    typeof body !== "string"
+  ) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  return selectArticles(articleId)
```

Comment:
In order to make sure the article exists, I'd recommend reading up on the notes. You will need to do this more than once _hint how can you do a 404 for comments??_ Consider a reusable function that checks if an article exists.